src/drive/web/modules/services/components/SuggestionProvider.jsx

```diff
@@ -46,44 +47,30 @@ class SuggestionProvider extends React.Component {
     )
   }
 
-  // fetches pretty much all the files and preloads FuzzyPathSearch
+  // fetches pretty much all the files not trashed and preloads FuzzyPathSearch
   async indexFiles() {
     const { client } = this.props
-    // TODO: fix me
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async resolve => {
-      const resp = await cozy.client.fetchJSON(
-        'GET',
-        `/data/io.cozy.files/_all_docs?include_docs=true`
-      )
-      const files = resp.rows
-        // TODO: fix me
-        // eslint-disable-next-line no-prototype-builtins
-        .filter(row => !row.doc.hasOwnProperty('views'))
-        .map(row => ({ id: row.id, ...row.doc }))
 
-      const folders = files.filter(file => file.type === TYPE_DIRECTORY)
+    const files = await client.queryAll(buildSuggestionsQuery().definition)
+
+    const folders = files.filter(file => file.type === TYPE_DIRECTORY)
 
-      const notInTrash = file =>
-        !file.trashed && !/^\/\.cozy_trash/.test(file.path)
-      const notOrphans = file =>
-        folders.find(folder => folder._id === file.dir_id) !== undefined
+    const notInTrash = file =>
+      !file.trashed && !/^\/\.cozy_trash/.test(file.path)
+    const notOrphans = file =>
+      folders.find(folder => folder._id === file.dir_id) !== undefined
 
-      const normalizedFilesPrevious = files
-        .filter(notInTrash)
-        .filter(notOrphans)
+    const normalizedFilesPrevious = files.filter(notInTrash).filter(notOrphans)
 
-      const normalizedFiles = await Promise.all(
-        normalizedFilesPrevious.map(
-          async file =>
-            await makeNormalizedFile(client, folders, file, getIconUrl)
-        )
+    const normalizedFiles = await Promise.all(
+      normalizedFilesPrevious.map(
+        async file =>
+          await makeNormalizedFile(client, folders, file, getIconUrl)
```

Comment:
This method can be pretty slow if we have a lot of Notes since it makes one call per note to get the url. Maybe we should fix that behavior too

Reply:
I think this is possible.
https://github.com/cozy/cozy-bar/blob/a61360a8d40f844102d9d12f4dd0ef9e5ae6bdba/src/components/SearchBar.jsx#L234
In order to make that, we could use onSelect as a function that will be called directly.

Reply:
On my instance (with a lot of notes coming from the team sharing) , I confirm the time taken by this method is huge and worst than the query itself: a search takes ~22s to complete, most of this time took by the notes queries. So it makes definitely sense to fix it

Reply:
done